frontend/composables/useRefBridge.ts

```diff
@@ -0,0 +1,112 @@
+ * */
+export function useObjectRefBridge<Torig, Tcreated> (origRef: Ref<Torig>|ModelRef<Torig>, origToCreated: ConverterCallback<Torig, Tcreated>, createdToOrig: ConverterCallback<Tcreated, Torig>) : Ref<Tcreated> {
+  const createdRef = ref<Tcreated>()
+  let pauseBack = false
+  let pauseForth = false
+  const stopperForth = watch(origRef, () => {
+    if (pauseForth) { return }
+    const OasC = (origRef.value !== undefined) ? origToCreated(origRef.value) : undefined
+    createdRef.value = OasC
+    pauseBack = true
+    nextTick(() => { pauseBack = false })
+  }, { immediate: true, deep: true })
+  const stopperBack = watch(createdRef, () => {
+    if (pauseBack) { return }
+    const CasO = (createdRef.value !== undefined) ? createdToOrig(createdRef.value) : undefined as unknown as Torig
+    origRef.value = CasO
+    pauseForth = true
+    nextTick(() => { pauseForth = false })
+  }, { deep: true })
+
+  onUnmounted(() => {
+    stopperBack()
+    stopperForth()
+  })
+
+  return createdRef as Ref<Tcreated>
+}
+
+/** This composable creates a two-way pipe between reactive variables of 2 different types. The values circulate back
+ *  and forth transparently from A to B and B to A, the reactivity is preserved on both ends, the types are converted
+ *  when the values pass through.
+ *  It solves 3 difficulties:
+ *  1. Different types. This is the main reason you need to bridge reactive variables: values of different types
+ *     that must stay synchronized.
+ *  2. Infinite loops. When one variable changes, the goal is to update the other one and you do not want that
+ *     the update triggers the first one again and so on.
+ *  3. At the moment the binding is created, it is typical that one variable has initial data and the other is
+ *     empty. You do not want the empty one to erase the initial data when the binding starts.
+ * @param origRef Ref that you want to bridge with the new ref that this function will create for you.
+ * @param origToCreated (optional) Callback/Arrow function that converts the value in the original ref into the type of the value in the created ref. If not provided, a basic conversion is performed, which is safe only between strings and numbers.
+ * @param createdToOrig (optional) Callback/Arrow function that converts the value in the created ref into the type of the value in the original ref. If not provided, a basic conversion is performed, which is safe only between strings and numbers.
+ * */
+export function usePrimitiveRefBridge<Torig, Tcreated> (origRef: Ref<Torig>|ModelRef<Torig>, origToCreated?: ConverterCallback<Torig, Tcreated>, createdToOrig?: ConverterCallback<Tcreated, Torig>) : Ref<Tcreated> {
+  return useObjectRefBridge<Torig, Tcreated>(origRef, origToCreated ?? convertPrimitiveByDefault, createdToOrig ?? convertPrimitiveByDefault)
+}
+
+function convertPrimitiveByDefault<TO> (from: any) : TO {
```

Comment:
would call `stringNumberConversion` that it is clear what it does, it would not be clear to me that the default mode would be to convert strings to number and vice versa.

Reply:
Thanks!
Done